elements/pl-latex-matrix-display/pl-latex-matrix-display.py

```diff
@@ -0,0 +1,56 @@
+import prairielearn as pl
+import lxml.html
+import numpy as np
+import chevron
+
+
+def prepare(element_html, element_index, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    required_attribs = ['params-name']
+    optional_attribs = ['label', 'digits', 'presentation-type']
+    pl.check_attribs(element, required_attribs, optional_attribs)
+
+
+def render(element_html, element_index, data):
+    element = lxml.html.fragment_fromstring(element_html)
+
+    # Get the number of digits to output
+    digits = pl.get_integer_attrib(element, 'digits', 2)
+    # Get the presentation type
+    presentation_type = pl.get_string_attrib(element, 'presentation-type', 'f')
+
+    var_name = pl.get_string_attrib(element, 'params-name')
+    # Get value of variable, raising exception if variable does not exist
+    var_data = data['params'].get(var_name, None)
+    if var_data is None:
+        raise Exception('No value in data["params"] for variable %s in pl-latex-matrix-display element' % var_name)
+
+    # If the variable is in a format generated by pl.to_json, convert it
+    # back to a standard type (otherwise, do nothing)
+    var_data = pl.from_json(var_data)
+
+    if np.isscalar(var_data):
+        raise Exception('Value in data["params"] for variable %s in pl-latex-matrix-display element must be an 1D or 2D array' % var_name)
+    else:
+        # Wrap the variable in an ndarray (if it's already one, this does nothing)
+        var_data = np.array(var_data)
+        # Check shape of variable
+        if var_data.ndim > 2:
+            raise Exception('Value in data["params"] for variable %s in pl-latex-matrix-display element must be an 1D or 2D array' % var_name)
+
+    # Get the label. Start with blank string if no label is given
+    label = pl.get_string_attrib(element, 'label', '')
+
+    # Create string for latex matrix format
+    latex_data = label + '$' + pl.latex_array_from_numpy_array(var_data, presentation_type=presentation_type, digits=digits) + '$'
```

Comment:
Would it be best if this element returned a string without `$` signs, for example so this string could be combined with other strings and all wrapped in something that may or may not be `$`? I'm confused, in particular, by suggesting the use of the label `$A = $` in the docs, which would result in:

`$A = $ $\begin{bmatrix} ... \end{bmatrix}$`

I would normally write:

`$A = \begin{bmatrix} ... \end{bmatrix}$`

Reply:
The goal of the element is to return a latex matrix (and not the string, that is done by the function `latex_from_2darray` (new name!)). If I don't wrap the variable `latex_data` with `$` sign, where else would I add `$`?


Someone may not want the label with latex fonts, and that is why I didn't add the label after the `$`. 

Please let me know if you had something specific in mind here. I can't think of a better solution at this time.

Reply:
I suggest removing the `$` signs. The user would then be responsible for wrapping the element in `$` signs for inline math or in `$$` signs for displayed math (https://docs.mathjax.org/en/latest/start.html#tex-and-latex-input).

I understand your concern about the label. Options include (1) removing the label, so requiring the user to prepend the element with a label of their own - this is how I would prefer to use the element; or (2) suggesting the syntax `\text{...}` if the user wants non-math content as the label. I'm sure there are other options.

Maybe another way to put this is that I would often use the element without the `$` signs, but would rarely use the element with the `$` signs.

What do you think?